pkg/vrct/vrctFs/merging.go

```diff
@@ -0,0 +1,196 @@
+package vrctFs
+
+import (
+	"fmt"
+	"gopkg.in/mgo.v2/bson"
+	"os"
+	"reflect"
+	"sort"
+)
+
+func (p *FilePrototype) mergeLayers() (PrototypeLayer, error) {
+	switch p.FileType {
+	case TextFile:
+		return p.mergeTextLayers()
+	default:
+		return p.mergeConfigLayers()
+	}
+}
+
+func (p *FilePrototype) mergeTextLayers() (PrototypeLayer, error) {
+	finalLayer := PrototypeLayer{
+		IsOptional: false,
+	}
+
+	finalContent, err := os.ReadFile(finalLayer.ContentPath)
+	if err != nil {
+		finalContent = nil
+	}
+
+	sort.Slice(p.Layers, func(i, j int) bool {
+		return !p.Layers[i].IsOptional && p.Layers[j].IsOptional
+	})
+
+	for i, currentLayer := range p.Layers {
+		currentContent, err := os.ReadFile(currentLayer.ContentPath)
+		if err != nil {
+			return finalLayer, err
+		}
+
+		if finalLayer.ContentPath != "" && currentLayer.ContentPath != "" && !currentLayer.IsOptional && !reflect.DeepEqual(finalContent, currentContent) {
+			return finalLayer, fmt.Errorf(
+				"non optional currentLayer nr. %d is in conflict with previous layers", i,
+			)
+		}
+		if (currentLayer.IsOptional && finalLayer.ContentPath == "") || !currentLayer.IsOptional {
+			finalLayer.ContentPath = currentLayer.ContentPath
+			finalContent = currentContent
+		}
+	}
+
+	return finalLayer, nil
+}
+
+func (p *FilePrototype) mergeConfigLayers() (PrototypeLayer, error) {
+	// create new layer first
+	finalLayer, err := p.CreateLayer(nil, nil, true)
```

Comment:
```suggestion
	finalLayer, err := p.CreateLayer(nil, nil, true)
```
![image](https://github.com/avorty/spito/assets/80683535/38460b70-dd3c-4649-813f-42866f107e70)